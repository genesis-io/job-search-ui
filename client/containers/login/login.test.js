import 'raf/polyfill';
import React from 'react';
import { shallow } from 'enzyme';
import { createMockStore } from 'redux-test-utils';
import Login from './login';

let component;
const testData = {
  validEmail: 'validemail@gmail.com',
  validPassword: 'password',
  invalidEmail: 'notavalidemail',
  invalidPassword: 'notvalid36#>!',
}
const testState = {
  user: {},
};
const store = createMockStore(testState);
const context = {
  store,
};

describe('Login Component, /login', () => {
  const wrappedComponent = () => {
    if (!component) {
      component = shallow(<Login location={{ pathname: '/login' }} />, { context }).dive({ context });
    }
    return component;
  }

  beforeEach(() => {
    component = undefined;
  });

  test('should render login component', () => {
    const wrapper = wrappedComponent();
    expect(wrapper).toMatchSnapshot();
  });

  test('should display error when email invalid on click', () => {
    const wrapper = wrappedComponent();
    wrapper.setState({
      email: testData.invalidEmail,
      password: testData.validPassword,
    });
    const button = wrapper.find('.login-form-sign-in');
    button.simulate('click', { preventDefault() {} });
    expect(wrapper.state().isEmailError).toBeTruthy();
    expect(wrapper.find('.invalid-form-error')).toHaveLength(1);
  });

  test('should display error when password invalid on click', () => {
    const wrapper = wrappedComponent();
    wrapper.setState({
      email: testData.validEmail,
      password: testData.invalidPassword,
    });
    const button = wrapper.find('.login-form-sign-in');
    button.simulate('click', { preventDefault() {} });
    expect(wrapper.state().isPasswordError).toBeTruthy();
    expect(wrapper.find('.invalid-form-error')).toHaveLength(1);
  });

  test('should clear form if no validation errors', () => {
    // const wrapper = wrappedComponent();
    // wrapper.setState({
    //   email: testData.validEmail,
    //   password: testData.validPassword,
    // });
    // expect(wrapper.state().email).toBeTruthy();
    // expect(wrapper.state().password).toBeTruthy();
    // const button = wrapper.find('.login-form-sign-in');
    // button.simulate('click', { preventDefault() {} });
    // expect(wrapper.state().email).toBeFalsy();
    // expect(wrapper.state().password).toBeFalsy();
  });

  test('should display error when email invalid on blur', () => {
    const wrapper = wrappedComponent();
    wrapper.setState({
      email: testData.invalidEmail,
      password: testData.validPassword,
    });
    const input = wrapper.find('[type="email"]');
    input.simulate('blur');
    expect(wrapper.state().isEmailError).toBeTruthy();
    expect(wrapper.find('.invalid-form-error')).toHaveLength(1);
  });

  test('should dispaly error when password invalid on blur', () => {
    const wrapper = wrappedComponent();
    wrapper.setState({
      email: testData.validEmail,
      password: testData.invalidPassword,
    });
    const input = wrapper.find('[type="password"]');
    input.simulate('blur');
    expect(wrapper.state().isPasswordError).toBeTruthy();
    expect(wrapper.find('.invalid-form-error')).toHaveLength(1);
  });

  test('should render button with (Log in) text', () => {
    const wrapper = wrappedComponent();
    expect(wrapper.find('.login-form-sign-in').text()).toEqual(' Log In');
  });

});


describe('Login Component, /signup', () => {
  const wrappedComponent = () => {
    if (!component) {
      component = shallow(<Login location={{ pathname: '/signup' }} />, { context }).dive({ context });
    }
    return component;
  }

  beforeEach(() => {
    component = undefined;
  });

  test('should render signp component', () => {
    const wrapper = wrappedComponent();
    expect(wrapper).toMatchSnapshot();
  });

  test('should render (Sign up with us! text)', () => {
    const wrapper = wrappedComponent();
    expect(wrapper.find('.login-form-header').text()).toEqual('Sign up with us!');
  });

  test('should render button with Register text', () => {
    const wrapper = wrappedComponent();
    expect(wrapper.find('.login-form-sign-in').text()).toEqual(' Register');
  });
});
