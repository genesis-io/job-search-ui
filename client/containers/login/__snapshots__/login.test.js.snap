// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component, /login should render login component 1`] = `
<div
  className="login-wrapper"
>
  <header
    className="login-wrapper-header"
  >
    Job Search
  </header>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <h3
      className="login-form-header"
    >
      Welcome Back
    </h3>
    <a
      className="default-button login-form-github"
      href="/api/auth/github"
    >
      <i
        className="fa fa-github button-icon"
      />
      <span>
         
        Login with GitHub
      </span>
    </a>
    <div
      className="login-form-or"
    >
      <hr />
      or
      <hr />
    </div>
    <label
      className="login-form-email"
    >
      Email Address
    </label>
    <input
      className=""
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <label>
      Password:
    </label>
    <input
      className=""
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <button
      className="default-button login-form-sign-in"
      onClick={[Function]}
      type="submit"
    >
       
      Log In
    </button>
  </form>
</div>
`;

exports[`Login Component, /signup should render signp component 1`] = `
<div
  className="login-wrapper"
>
  <header
    className="login-wrapper-header"
  >
    Job Search
  </header>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <h3
      className="login-form-header"
    >
      Sign up with us!
    </h3>
    <a
      className="default-button login-form-github"
      href="/api/auth/github"
    >
      <i
        className="fa fa-github button-icon"
      />
      <span>
         
        Signup with Github
      </span>
    </a>
    <div
      className="login-form-or"
    >
      <hr />
      or
      <hr />
    </div>
    <label
      className="login-form-email"
    >
      Email Address
    </label>
    <input
      className=""
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <label>
      Password:
    </label>
    <input
      className=""
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <button
      className="default-button login-form-sign-in"
      onClick={[Function]}
      type="submit"
    >
       
      Register
    </button>
  </form>
</div>
`;
